lib/features/support/app-automate.rb

```diff
@@ -0,0 +1,63 @@
+require 'appium_lib'
+require_relative './driver'
+
+class AppAutomateDriver < Driver
+
+  attr_reader :device_type
+
+  APP_UPLOAD_URI = "https://api-cloud.browserstack.com/app-automate/upload"
+
+  def devices
+    Devices::DEVICE_HASH
+  end
+
+  def start_driver(target_device, app_location)
+    @device_type = target_device
+    upload_app(app_location)
+    start_local_tunnel
+    @capabilities.merge! devices[target_device]
+    @driver = Appium::Driver.new({
+      'caps' => @capabilities,
+      'appium_lib' => {
+        :server_url => "http://#{@username}:#{@access_key}@hub-cloud.browserstack.com/wd/hub"
+      }
+    }, false).start_driver
+  end
+
+  def stop_driver
```

Comment:
What is this method called by?

Reply:
This can be called by the local environment of whichever feature set is running.  There's a `@driver.quit` call added to `at_exit` in the `Driver` initialisation, but it may be required by certain feature sets (although it can probably be removed and re-added later if necessary).